build.sh

```diff
@@ -1,13 +1,22 @@
 #!/bin/bash
 
-echo $1
-gomod=$(go list -m -json github.com/golangci/golangci-lint@$1 | grep -o '"GoMod": "[^"]*' | grep -o '[^"]*$')
-output=$(cat $gomod)
+echo "$1"
+gomod=$(go list -m -json github.com/golangci/golangci-lint@"$1" | grep -o '"GoMod": "[^"]*' | grep -o '[^"]*$')
+output=$(cat "$gomod")
 tools_version=$(grep 'golang.org/x/tools' <<< "$output" | grep -oE '\bv[0-9]+\.[0-9]+\.[0-9]+\b' | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+go_version=$(grep -oE '^go [0-9]+\.[0-9].+' <<< "$output" | cut -d ' ' -f2)
+
+local_go_version=$(go version | grep -oE '[0-9]+\.[0-9]+\.[0-9]+')
+if [[ "$go_version" != "$local_go_version" ]]; then
```

Comment:
Reason for adding this if: it would be good to know that there is a version mismatch, i.e. I am using Go 1.21 and golangci-lint v1.56.X is using Go 1.22. 

It might not make a difference so this just outputs error messages and does not stop the build.